### ISSUE TITLE
Temperature displays in Celsius only after latest update

### ISSUE THREAD

**Issue by boctok** [georgezhao2010/midea_ac_lan#292]
### HA core version

2023.8.4

### Intergration version

 v0.3.19

### Device type and model

Air Conditioner 00000Q1D

### Used App

Midea Air

### The description of problem

I have two AC units. After the latest update, one of the units was "unavailable" and the other one now displays the temperature in Celsius. I managed to add the other unit back in, but they both display the temperature in Celsius and there doesn't seem to be a way to enable temperature display in Fahrenheit. They are both configured to show the temperature in Fahrenheit in the Midea app.

### The logs

Nothing in the logs about displaying the temperature in the wrong units.

**Comment by georgezhao2010** [comment]
> ### The description of problem
> I have two AC units. After the latest update, one of the units was "unavailable" and the other one now displays the temperature in Celsius. I managed to add the other unit back in, but they both display the temperature in Celsius and there doesn't seem to be a way to enable temperature display in Fahrenheit. They are both configured to show the temperature in Fahrenheit in the Midea app.
> 
Could you eable the debug log and show me a log containing ‘body type’: ‘c0’, which looks like the following

```
Received: {'header': 'aa2aac00000000000803', 'body': 'c0004a667f7f00000004006aff0e000200000000000000000000000019ff5224', 'message type': '03', 'body type': 'c0'}
```

**Comment by georgezhao2010** [comment]
> Yes, it displayed the temperatures in Fahrenheit before.
> 
> `Received: {'header': 'aa23ac00000000000303', 'body': 'c00125660000003c0010045d5b000000000000000000001aa8', 'message type': '03', 'body type': 'c0'}`

OK, I got it, thank you.

For a temporary solution, you can open the file `custom_components/midea_ac_lan/climate.py in your homeassistant`, replace the line 68 `return TEMP_FAHRENHEIT if self._device.get_attribute("temp_fahrenheit") else TEMP_CELSIUS` with `return TEMP_CELSIUS`, then makr true the temperatures displayed in Fahrenhei. 

replace
```Python
    @property
    def temperature_unit(self):
        return TEMP_FAHRENHEIT if self._device.get_attribute("temp_fahrenheit") else TEMP_CELSIUS
```
with
```Python
    @property
    def temperature_unit(self):
        return TEMP_CELSIUS
```

**Comment by boctok** [comment]
Not sure where I need to set the temperatures displayed in Fahrenheit to true. In the app? They are already set to °F
After changing the code, I can see no change. It still displays the numerical temperature in Celsius, but with °F after the number, like it was before. 
![Midea card](https://github.com/georgezhao2010/midea_ac_lan/assets/5306739/c899b002-60a7-447a-bdaa-8be2ce58370f)

It is obviously not 21°F in my house.

**Comment by georgezhao2010** [comment]
> Not sure where I need to set the temperatures displayed in Fahrenheit to true. In the app? They are already set to °F After changing the code, I can see no change. It still displays the numerical temperature in Celsius, but with °F after the number, like it was before. 
> 
> It is obviously not 21°F in my house.

I only need to specify whether the temperature is in Fahrenheit or Celsius through the function, and HA will handle the rest of the numerical conversion based on your temperature unit settings. Additionally, whether the symbol following the number is Celsius or Fahrenheit is also determined by your temperature unit settings.

It is in `Settings -> System -> General`

After reviewing your log, I have clearly informed HA through the previous modification that the number 20 or 21 is in Celsius, and HA should have automatically converted it to Fahrenheit. Are you certain that you made the modification correctly? Have you restarted HA? How about trying to delete the device and adding it back again?

**Comment by WeaselBreath** [comment]
ditto, same issue here w/ two Toshiba ac units

**Comment by Xe138** [comment]
Same issue.

![image](https://github.com/georgezhao2010/midea_ac_lan/assets/50377201/cf9f8a6d-b818-4059-97f8-e636daee54c9)

**Comment by ghgeiger** [comment]
I am having the same problem. When I look at the file climate.py, you suggest to replace line 68, but the referenced code seems to be line 84 in my version. I think I'll just wait for a fix.

**Comment by boctok** [comment]
Mine reverted back to °F after several restarts with the code change. Maybe it was stuck in my browser cache.

**Comment by WeaselBreath** [comment]
confirm fix also works for me on line 84 after hard reboot

**Comment by georgezhao2010** [comment]
> I am having the same problem. When I look at the file climate.py, you suggest to replace line 68, but the referenced code seems to be line 84 in my version. I think I'll just revert, and wait for a fix.
> 
> Edit just to clarify, when using the 'file edit' add-on in home assistant, what you refer to as line 68 appears as line 84. That may just be how that application formats things, I'm not sure.

I'm sorry, it should be line 84.

**Comment by georgezhao2010** [comment]
> 



> Mine reverted back to °F after several restarts with the code change and, yes, it is line 84, not 68. Maybe it was stuck in my browser cache.

Sorry, it should be line 84

**Comment by SkinnyT81** [comment]
This temporary fix worked for me as well.

**Comment by Medic97** [comment]
This only worked for the main widget, not for the inside and outside temp widgets. 
<img width="273" alt="Screenshot 2023-09-05 at 11 15 25 AM" src="https://github.com/georgezhao2010/midea_ac_lan/assets/23649478/f53a6194-56c5-4e93-90fc-8d23836ac59a">
<img width="271" alt="Screenshot 2023-09-05 at 11 15 29 AM" src="https://github.com/georgezhao2010/midea_ac_lan/assets/23649478/a1f38663-c804-4ecc-b08b-1ef6e1f23fc1">
<img width="282" alt="Screenshot 2023-09-05 at 11 15 33 AM" src="https://github.com/georgezhao2010/midea_ac_lan/assets/23649478/98ddd760-6dde-4f30-b479-fc451b54fdcf">

**Comment by georgezhao2010** [comment]
Fixed in v0.3.20
